app/main.py

```diff
@@ -1,3 +1,122 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(
+        self, other: Union["Distance", int, float]
+    ) -> Union["Distance", TypeError]:
```

Comment:
I don't know how it works with newer version of Python but on my 3.11 it didn't let me write ```other: Distance``` only ```other: "Distance"``` and after that if I wanted to use ```other: "Distance" | int | float``` tests were giving me an error, that's why I had to use ```Union```.

Reply:
You can use import annotation from __future__ in this case